File: 2020-advent-of-code/1/1.go
package main

import (
	"fmt"
	"io/ioutil"
	"strconv"
	"strings"
)

func check(e error) {
	if e != nil {
		panic(e)
	}
}

func main() {
	dat, err := ioutil.ReadFile("input.txt")
	check(err)

	lines := strings.Split(string(dat), "\n")

	var nums []int64
	for _, line := range lines {
		num, err := strconv.ParseInt(line, 0, 64)
		if err == nil {
			nums = append(nums, num)
		}
	}

	for i := 0; i < (len(nums) - 1); i++ {
		for j := i; j < len(nums); j++ {
			if nums[i]+nums[j] == 2020 {
				fmt.Println(nums[i] * nums[j])
				return
			}
		}
	}
}

File: 2020-advent-of-code/1/1.rs
fn main() {
    let dat = std::fs::read_to_string("input.txt").expect("couldn't read file");

    let nums: Vec<i64> = dat
        .lines()
        .map(|x| x.parse::<i64>().expect("invalid int"))
        .collect();

    for i in 0..(nums.len() - 1) {
        for j in i..nums.len() {
            if nums[i] + nums[j] == 2020 {
                println!("{}", nums[i] * nums[j]);
                return;
            }
        }
    }
}
